vm-setup/roles/v1aX_integration_test/tasks/move.yml

```diff
@@ -95,53 +107,49 @@
 
   # Check for pods & nodes on the target cluster
   - name: Check if pods in running state
-    shell: "kubectl get pods -A -o json | jq -r '.items[].status.phase' | grep -v Running"
-    environment:
-      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
-    retries: 150
+    k8s_info:
+      kind: pods 
+      kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+      field_selectors:
+        - status.phase!=Running
+    register: target_running_pods 
+    retries: 150 
     delay: 20
-    register: target_running_pods
-    failed_when: >
-      (target_running_pods.stderr != "") or
-      (target_running_pods.rc > 1) or
-      (target_running_pods.stdout != "")
-    until: target_running_pods.stdout == ""
-
+    until: target_running_pods.resources | length == 0
+    failed_when: target_running_pods.resources | length > 0 
```

Comment:
Does this one line of failed_when simulates the three lines removed above?

Reply:
IMHO the shell was executing `kubectl get pods -A -o json` then filters phase and finally `grep` lines that do not contain the `Running` word. As a result the `stdout` will contain lines of non running phases or `""` if all pods are in `Running` phase.

So, the deleted failed_when was checking if `target_running_pods.stdout != ""` and this is handled by `target_running_pods.resources | length == 0` the other `target_running_pods.stderr != "" `and `target_running_pods.rc > 1` As far as I can tell are shell cmd related and this should be handled by the module `k8s_info`

Docs:
- stderr: https://docs.ansible.com/ansible/latest/reference_appendices/common_return_values.html#stderr
- rc: https://docs.ansible.com/ansible/latest/reference_appendices/common_return_values.html#rc